cloud-annotations/src/main/java/cloud/commandframework/annotations/FlagExtractor.java

```diff
@@ -78,14 +80,25 @@
                                     parameter.getType().getCanonicalName(), flag.value(), method.getName()
                             ));
                 }
-                final CommandArgument.Builder argumentBuilder = CommandArgument.ofType(
+                final BiFunction<?, @NonNull String, @NonNull List<String>> suggestionProvider;
+                if (!flag.suggestions().isEmpty()) {
+                    suggestionProvider = registry.getSuggestionProvider(flag.suggestions()).orElse(null);
+                } else {
+                    suggestionProvider = null;
+                }
+                final CommandArgument.Builder argumentBuilder0 = CommandArgument.ofType(
                         parameter.getType(),
                         flag.value()
                 );
-                final CommandArgument argument = argumentBuilder.asRequired()
+                final CommandArgument.Builder argumentBuilder = argumentBuilder0.asRequired()
                         .manager(this.commandManager)
-                        .withParser(parser)
-                        .build();
+                        .withParser(parser);
+                final CommandArgument argument;
+                if (suggestionProvider != null) {
+                    argument = argumentBuilder.withSuggestionsProvider(suggestionProvider).build();
```

Comment:
Would it be an idea to just have `withSuggestionsProvider` accept a `null` for default?